lib/Support/TimeProfiler.cpp

```diff
@@ -115,23 +122,23 @@ struct TimeTraceProfiler {
     // Emit totals by section name as additional "thread" events, sorted from
     // longest one.
     int Tid = 1;
-    std::vector<NameAndDuration> SortedTotals;
-    SortedTotals.reserve(TotalPerName.size());
-    for (const auto &E : TotalPerName) {
-      SortedTotals.push_back(E);
-    }
+    std::vector<NameAndCountAndDurationType> SortedTotals;
```

Comment:
Are all these changes required to add the granularity flag?  Kind of surprised at the amount of churn here to turn a hard coded 500 into something controlled by a command line option.

Reply:
I followed the changes in llvm-project.  The Granularity setting was added... but not connnected initially to any external configuration.  Then there were follow up PRs that added the ftime-trace-granularity option to the compiler to configure that.  That is where these changes come in.  The way options are connected differs in the llvm-project branch the the DXC branch, to I had to follow how DXC wires in the option.

Reply:
I think all of the defaulting to 500us might be overkill.  I am open to better suggestions here.  A default is required if the option is not specified.

Reply:
If that's hard to do then maybe just making the changes to the granularity checking and not modifying anything else would reduce the risk of this.

Reply:
My point is that this commit is doing much more than just adding that option.